ArduPlane/takeoff.cpp

```diff
@@ -400,3 +401,17 @@ bool Plane::check_takeoff_timeout(void)
      return false;
 }
 
+/*
+ check if the pitch level-off time has expired; returns true if timeout has occurred
+*/
+bool Plane::check_takeoff_timeout_level_off(void)
+{
+    if (takeoff_state.level_off_start_time_ms > 0) {
+        // A takeoff is in progress.
+        uint32_t now = AP_HAL::millis();
+        if ((now - takeoff_state.level_off_start_time_ms) > (uint32_t)(1000U * g.takeoff_pitch_limit_reduction_sec)) {
```

Comment:
@peterbarker is this the right way to check a timeout in the face of time folding?